[BUG] - Error in build of latest release 1.35.4

*External* 

**Area**
cardano-node 1.35.4

**Summary**
After running 
```cabal update
cabal build all 
```
`Failed to build HsOpenSSL-0.11.7.3` occurs and build fails

Error Details
```
Failed to build HsOpenSSL-0.11.7.3.
Build log (
/home/usr/.cabal/logs/ghc-8.10.7/HsOpenSSL-0.11.7.3-2fbf460de846e6520ee3ac1449f168f6540c4acba96ca535dbe38914e6432166.log
):
[1 of 1] Compiling Main             ( /home/usr/git/cardano-node2/dist-newstyle/tmp/src-2737816/HsOpenSSL-0.11.7.3/dist/setup/setup.hs, /home/usr/git/cardano-node2/dist-newstyle/tmp/src-2737816/HsOpenSSL-0.11.7.3/dist/setup/Main.o )
Linking /home/usr/git/cardano-node2/dist-newstyle/tmp/src-2737816/HsOpenSSL-0.11.7.3/dist/setup/setup ...
Configuring HsOpenSSL-0.11.7.3...
Preprocessing library for HsOpenSSL-0.11.7.3..
Building library for HsOpenSSL-0.11.7.3..
[ 1 of 33] Compiling OpenSSL          ( OpenSSL.hs, dist/build/OpenSSL.o, dist/build/OpenSSL.dyn_o )
[ 2 of 33] Compiling OpenSSL.DH.Internal ( OpenSSL/DH/Internal.hs, dist/build/OpenSSL/DH/Internal.o, dist/build/OpenSSL/DH/Internal.dyn_o )
[ 3 of 33] Compiling OpenSSL.ERR      ( OpenSSL/ERR.hs, dist/build/OpenSSL/ERR.o, dist/build/OpenSSL/ERR.dyn_o )
[ 4 of 33] Compiling OpenSSL.EVP.Base64 ( OpenSSL/EVP/Base64.hs, dist/build/OpenSSL/EVP/Base64.o, dist/build/OpenSSL/EVP/Base64.dyn_o )
[ 5 of 33] Compiling OpenSSL.Objects  ( dist/build/OpenSSL/Objects.hs, dist/build/OpenSSL/Objects.o, dist/build/OpenSSL/Objects.dyn_o )
[ 6 of 33] Compiling OpenSSL.SSL.Option ( dist/build/OpenSSL/SSL/Option.hs, dist/build/OpenSSL/SSL/Option.o, dist/build/OpenSSL/SSL/Option.dyn_o )
[ 7 of 33] Compiling OpenSSL.Stack    ( dist/build/OpenSSL/Stack.hs, dist/build/OpenSSL/Stack.o, dist/build/OpenSSL/Stack.dyn_o )
[ 8 of 33] Compiling OpenSSL.Utils    ( OpenSSL/Utils.hs, dist/build/OpenSSL/Utils.o, dist/build/OpenSSL/Utils.dyn_o )
[ 9 of 33] Compiling OpenSSL.Random   ( OpenSSL/Random.hs, dist/build/OpenSSL/Random.o, dist/build/OpenSSL/Random.dyn_o )
/tmp/ghc2765403_0/ghc_123.c: In function ‘ghczuwrapperZC1ZCHsOpenSSLzm0zi11zi7zi3zm2fbf460de846e6520ee3ac1449f168f6540c4acba96ca535dbe38914e6432166ZCOpenSSLziRandomZCRANDzupseudozubytes’:

＊snip＊

`gcc' failed in phase `C Compiler'. (Exit code: 1)
cabal: Failed to build HsOpenSSL-0.11.7.3 (which is required by
test:cardano-node-test from cardano-node-1.35.4, exe:cardano-node from
cardano-node-1.35.4 and others). See the build log above for details.
```

**Points of concern**
HsOpenSSL-0.11.7.3 in the Haskell package released on the 17th has a build error to begin with.
https://hackage.haskell.org/package/HsOpenSSL-0.11.7.3

**Steps to reproduce**
Steps to reproduce the behavior:
```
git clone https://github.com/input-output-hk/cardano-node.git cardano-node
cd cardano-node
cabal update
git fetch --all --recurse-submodules --tags
git checkout tags/1.35.4
cabal configure -O0 -w ghc-8.10.7
echo -e "package cardano-crypto-praos\n flags: -external-libsodium-vrf" > cabal.project.local
cabal build all 
```
**Expected behavior**
A clear and concise description of what you expected to happen.

**System info (please complete the following information):**
- OS Name: Ubuntu
- OS Version 20.04

Encountered similar bug on Fedora 36 OS. After I manually install `ghc-HsOpenSSL-devel` I got a long list of errors. Here are just 2: 
```
Building library for HsOpenSSL-0.11.7.3..
[ 1 of 33] Compiling OpenSSL          ( OpenSSL.hs, dist/build/OpenSSL.o, dist/build/OpenSSL.dyn_o )
[ 2 of 33] Compiling OpenSSL.DH.Internal ( OpenSSL/DH/Internal.hs, dist/build/OpenSSL/DH/Internal.o, dist/build/OpenSSL/DH/Internal.dyn_o )
/tmp/ghc28285_0/ghc_38.c: In function ‘ghczuwrapperZC0ZCHsOpenSSLzm0zi11zi7zi3zm2fbf460de846e6520ee3ac1449f168f6540c4acba96ca535dbe38914e6432166ZCOpenSSLziDHziInternalZCDHzufree’:

/tmp/ghc28285_0/ghc_38.c:9:1: error:
     warning: ‘DH_free’ is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
        9 | void ghczuwrapperZC0ZCHsOpenSSLzm0zi11zi7zi3zm2fbf460de846e6520ee3ac1449f168f6540c4acba96ca535dbe38914e6432166ZCOpenSSLziDHziInternalZCDHzufree(DH* a1) {DH_free(a1);}
          | ^~~~
  |
9 | void ghczuwrapperZC0ZCHsOpenSSLzm0zi11zi7zi3zm2fbf460de846e6520ee3ac1449f168f6540c4acba96ca535dbe38914e6432166ZCOpenSSLziDHziInternalZCDHzufree(DH* a1) {DH_free(a1);}
  | ^

In file included from /tmp/ghc28285_0/ghc_38.c:7:0: error: 

/usr/include/openssl/dh.h:200:28: error:
     note: declared here
      200 | OSSL_DEPRECATEDIN_3_0 void DH_free(DH *dh);
          |                            ^~~~~~~
    |
200 | OSSL_DEPRECATEDIN_3_0 void DH_free(DH *dh);
    |                            ^
```

--- a/cabal.project
+++ b/cabal.project
@@ -80,7 +80,7 @@ constraints:
     -- systemd-2.3.0 requires at least network 3.1.1.0 but it doesn't declare
     -- that dependency
   , network >= 3.1.1.0
-  , HSOpenSSL >= 0.11.7.2
+  , HsOpenSSL == 0.11.7.2
   , algebraic-graphs < 0.7
   , protolude >= 0.3.2
 

